### ISSUE TITLE
required missing property when trying to attach image inline

### ISSUE THREAD

**Issue by ajwillo** [O365/python-o365#262]
when attempting to send a message with an inline image, the below error occurs. when I remove the code for the image/attachment the mail sends

```
    account = Account((credentials[0]['username'], credentials[0]['password']))
    mailbox = account.mailbox()
    message = mailbox.new_message()
    message.attachments.add('https://www.image.com/email_banner.png')
    att = message.attachments[0]  # get the attachment object
    att.is_inline = True
    att.content_id = 'banner.png'
    body += '<p><img src="cid:banner.png"></p>'
    message.subject = 'Hello Test'
    message.to.add(['user@user.com']) 
    message.body = body
    message.send()
```
error
```
Client Error: 400 Client Error: Bad Request for url: https://graph.microsoft.com/v1.0/me/sendMail | Error Message: Required property is missing.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/O365/message.py", line 602, in send
    response = self.con.post(url, data=data)
  File "/usr/local/lib/python3.6/site-packages/O365/connection.py", line 684, in post
    return self.oauth_request(url, 'post', data=data, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/O365/connection.py", line 662, in oauth_request
    return self._internal_request(self.session, url, method, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/O365/connection.py", line 627, in _internal_request
    raise HTTPError('{} | Error Message: {}'.format(e.args[0], error_message), response=response) from None
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://graph.microsoft.com/v1.0/me/sendMail | Error Message: Required property is missing.    
```

**Comment by janscas** [comment]
The attachment must be an image, not a url.

**Comment by janscas** [comment]
Can we close this?

**Comment by ajwillo** [comment]
I've used pillow to load the image but am still unable to attach

```
>>>img = Image.open(BytesIO(image_url.content))
>>>message.attachments.add(img)
ValueError: Attachments must be a str or Path or a list, tuple or set of the former
```

**Comment by janscas** [comment]
For the moment you can’t attach in memory objects. Save the file to disk and the upload and delete the file after.

I’ll try to add this.

**Comment by janscas** [comment]
This will be released under version 2.0.

You can now use inmemory objects:

```python
response = requests.get('https://www.image.com/email_banner.png')
image = BytesIO(response.content) # In memory

# ...
msg = account.new_message()
msg.attachments.add([(image, 'my_file_image.png')])  # pass a list of tuples: (BytesIO instance, file_name)
att = msg.attachments[0]  # get the attachment object
att.is_inline = True
att.content_id = 'banner.png'
body = '<p><img src="cid:banner.png"></p>'
msg.subject = 'Hello Test'
msg.to.add(['user@user.com']) 
msg.body = body
msg.send()
```

**Comment by davidlpotter59** [comment]
I am having issues with the 0365 module.  I authenticate perfectly.   The errors comes into play at the account objects.   

The message I get is: Client Error: 400 Client Error: Bad Request for url: https://graph.microsoft.com/v1.0/sendMail | Error Message: Resource not found for the segment 'sendMail'.
Traceback (most recent call last):
  File "mail5.py", line 18, in <module>
    message.send()
  File "/home/davep/.local/share/virtualenvs/completed-88qwCt9r/lib/python3.7/site-packages/O365/message.py", line 654, in send
    response = self.con.post(url, data=data)
  File "/home/davep/.local/share/virtualenvs/completed-88qwCt9r/lib/python3.7/site-packages/O365/connection.py", line 718, in post
    return self.oauth_request(url, 'post', data=data, **kwargs)
  File "/home/davep/.local/share/virtualenvs/completed-88qwCt9r/lib/python3.7/site-packages/O365/connection.py", line 696, in oauth_request
    return self._internal_request(self.session, url, method, **kwargs)
  File "/home/davep/.local/share/virtualenvs/completed-88qwCt9r/lib/python3.7/site-packages/O365/connection.py", line 658, in _internal_request
    raise HTTPError('{} | Error Message: {}'.format(e.args[0], error_message), response=response) from None
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://graph.microsoft.com/v1.0/sendMail | Error Message: Resource not found for the segment 'sendMail'.

Code is:
from O365 import Account

credentials = ('ID', 'secret')

account = Account(credentials, auth_flow_type='credentials')
if account.authenticate():
   print('Authenticated!') # this prints

mailbox = account.mailbox()
message = mailbox.new_message()
message.subject = 'Hello Test'
message.to.add(['user@somedomain.com']) 
message.body = "You have documents"
message.send()

Thank you for anyone that can help me.
Versions: python 3.6.8, O365 2.0.3
